tests/agent-opentracing/src/test/java/org/hawkular/apm/tests/agent/opentracing/client/http/ApacheHttpClientITest.java

```diff
@@ -257,12 +281,17 @@ public String handleResponse(
 
         Producer testProducer = producers.get(0);
 
-        String path = URI.create(SAY_HELLO_URL).getPath();
+        if (!badUrl) {
+            String path = request.getURI().getPath();
 
-        assertEquals(path, testProducer.getUri());
+            assertEquals(path, testProducer.getUri());
```

Comment:
Shouldn't be `testProducer.getUri()` always equal to `request.getURI().getPath()`?

Currently for a valid request it is request path and for bad url it is the whole URL?